lib/diaspora_federation/properties_dsl.rb

```diff
@@ -71,11 +71,18 @@ def class_prop_names
     def define_property(name, type, opts={})
       raise InvalidName unless name_valid?(name)
 
-      xml_name = name
       if opts.has_key? :xml_name
         raise ArgumentError, "xml_name is not supported for nested entities" unless type == String
         xml_name = opts[:xml_name]
         raise InvalidName, "invalid xml_name" unless name_valid?(xml_name)
+      elsif type == String
+        xml_name = name
+      elsif type.instance_of?(Array)
+        xml_name = type.first.entity_name.to_sym
+      elsif type.ancestors.include?(Entity)
+        xml_name = type.entity_name.to_sym
+      else
+        raise ArgumentError, "unknown type #{type} supplied"
```

Comment:
> Well, that's kinda philosophical. I personally prefer to keep public API surface as low as possible, allowing for easier and faster iteration without breaking other parts. Hence I prefer private instance methods if they're used only within the class to extract redundant logic, even if it doesn't depend on instance data.

I agree with @jhass here. I try to make as much private as possible, we can always make something public if we need it. And I only do class methods if I need it (for example if they are called from another public or private class method)